ams/api/src/main/java/com/netease/arctic/ams/api/metrics/SelfOptimizingReport.java

```diff
@@ -0,0 +1,130 @@
+import com.codahale.metrics.Gauge;
+
+public class SelfOptimizingReport implements MetricReport {
+  public static final String ICEBERG_COMMIT_REPORT_NAME = "self_optimizing_report";
+
+  public static final String TABLE_NAME = "table-name";
+  public static final String OPTIMIZING_PROCESS_ID = "optimizing-process-id";
+  public static final String OPTIMIZING_TYPE = "optimizing-type";
+  public static final String TARGET_SNAPSHOT_ID = "target-snapshot-id";
+  public static final String OPTIMIZING_STATUS = "optimizing-status";
+
+  private static final String TABLE_OPTIMIZING_TOTAL_COST_TIME_SECOND = "table-optimizing-total-cost-time-second";
+  private static final String TABLE_OPTIMIZING_PLAN_DURATION_SECOND = "table-optimizing-plan-duration-second";
+  private static final String TABLE_OPTIMIZING_IDLE_DURATION_SECOND = "table-optimizing-idle-duration-second";
+  private static final String TABLE_OPTIMIZING_PENDING_DURATION_SECOND = "table-optimizing-pending-duration-second";
+  private static final String TABLE_OPTIMIZING_COMMIT_DURATION_SECOND = "table-optimizing-commit-duration-second";
+
+  private final String tableName;
+  private Long optimizingProcessId;
+  private String optimizingType;
+  private Long targetSnapshotId;
+  private String optimizingStatus;
+
+  private final Counter tableOptimizingTotalCostTimeMillisecond = new Counter();
+  private final Counter tableOptimizingPlanDurationSecond = new Counter();
+  private final Counter tableOptimizingIdleDurationSecond = new Counter();
+  private final Counter tableOptimizingPendingDurationSecond = new Counter();
+  private final Counter tableOptimizingCommitDurationSecond = new Counter();
+
+  public SelfOptimizingReport(String tableName) {
+    this.tableName = tableName;
+  }
+
+  public void setOptimizingProcessId(Long optimizingProcessId) {
+    this.optimizingProcessId = optimizingProcessId;
+  }
+
+  public void setOptimizingType(String optimizingType) {
+    this.optimizingType = optimizingType;
+  }
+
+  public void setTargetSnapshotId(Long targetSnapshotId) {
+    this.targetSnapshotId = targetSnapshotId;
+  }
+
+  public void setOptimizingStatus(String optimizingStatus) {
+    this.optimizingStatus = optimizingStatus;
+  }
+
+  @Override
+  public String name() {
+    return ICEBERG_COMMIT_REPORT_NAME;
+  }
+
+  @TaggedMetrics.Tag(name = TABLE_NAME)
+  public String tableName() {
+    return tableName;
+  }
+
+  @TaggedMetrics.Tag(name = OPTIMIZING_PROCESS_ID)
+  public Long optimizingProcessId() {
+    return optimizingProcessId;
+  }
+
+  @TaggedMetrics.Tag(name = OPTIMIZING_TYPE)
+  public String optimizingType() {
+    return optimizingType;
+  }
+
+  @TaggedMetrics.Tag(name = TARGET_SNAPSHOT_ID)
+  public Long targetSnapshotId() {
+    return targetSnapshotId;
+  }
+
+  @TaggedMetrics.Tag(name = OPTIMIZING_STATUS)
+  public String optimizingStatus() {
+    return optimizingStatus;
+  }
+
+
+  @TaggedMetrics.Metric(name = TABLE_OPTIMIZING_TOTAL_COST_TIME_SECOND)
```

Comment:
Is it same with the cost unit (ms) in the existing logic?

Reply:
The TABLE_OPTIMIZING_TOTAL_COST_TIME_SECOND indicator should collect the sum of all task executions, because the unit is seconds, so I want to count ms first and then return the sum, so here I will use ms as the unit to collect each task, but in the end the indicators are all added up and returned.

But it would be better to use `Timer` to collect here, I will optimize this part